reflectable/lib/src/builder_implementation.dart

```diff
@@ -5005,7 +5005,18 @@ Future<String> _extractMetadataCode(Element element, Resolver resolver,
       if (_isPrivateName(name)) {
         await _severe('Cannot access private name $name', element);
       }
-      metadataParts.add('const $prefix$name($arguments)');
+      if (annotationNode.typeArguments != null) {
+        String typeArguments = annotationNode.typeArguments!.arguments
+            .map((TypeAnnotation typeArgument) {
+              LibraryElement library = typeArgument.type!.element!.library!;
+              String prefix = importCollector._getPrefix(library);
```

Comment:
For the prefix we need to check that the given library is importable (`_isImportable`). For instance, `dart:_internal` is not importable by user code.

Reply:
I think I've resolved this, I currently have the type argument default to `'dynamic'` if the library can't be imported, though I'm not sure if this is the correct behavior.